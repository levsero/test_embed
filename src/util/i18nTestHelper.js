import { i18n } from 'service/i18n';
import { createStore } from 'redux';
import reducer from 'src/redux/modules/reducer';
import yaml from 'js-yaml';
import fs from 'fs';
import path from 'path';

i18n.init(createStore(reducer));

// Load fallback translations for tests
const oldVal = global.__DEV__;
const fallbackPath = path.resolve(__dirname, '../../config/locales/translations/embeddable_framework.yml');
const fallbackTranslations = yaml.safeLoad(fs.readFileSync(fallbackPath));

global.__DEV__ = true;
i18n.setFallbackTranslations(fallbackTranslations);
global.__DEV__ = oldVal;

i18n.setLocale('en-US');
