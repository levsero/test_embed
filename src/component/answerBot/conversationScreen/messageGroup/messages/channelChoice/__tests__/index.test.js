import 'utility/i18nTestHelper';
import { render, fireEvent } from 'react-testing-library';
import React from 'react';

import { PureChannelChoice as ChannelChoice } from '../index';

const actions = Object.freeze({
  updateBackButtonVisibility: jest.fn(),
  updateActiveEmbed: jest.fn()
});

const renderComponent = (props = {}) => {
  const defaultProps = {
    callbackAvailable: false,
    talkAvailable: false,
    submitTicketAvailable: false,
    chatAvailable: false,
    actions
  };

  const componentProps = {
    ...defaultProps,
    ...props
  };

  return render(<ChannelChoice {...componentProps} />);
};

test('renders the expected classes', () => {
  const { container } = renderComponent({
    callbackAvailable: true,
    talkAvailable: true,
    submitTicketAvailable: true,
    chatAvailable: true
  });

  expect(container)
    .toMatchSnapshot();
});

test('renders the leading message', () => {
  const { container } = renderComponent({
    leadingMessage: 'hello world',
    callbackAvailable: true,
    talkAvailable: true,
    submitTicketAvailable: true,
    chatAvailable: true
  });

  expect(container)
    .toMatchSnapshot();
});

describe('channels', () => {
  const assertExpected = (props) => {
    const { container } = renderComponent(props);

    expect(container)
      .toMatchSnapshot();
  };

  it('renders the passed in leading message regardless of other properties', () => {
    assertExpected({
      submitTicketAvailable: true,
      leadingMessage: 'blah blah'
    });
  });

  it('renders expected leading message if only submit ticket is available', () => {
    assertExpected({ submitTicketAvailable: true, leadingMessage: 'blah' });
  });

  it('renders expected leading message if only chat is available', () => {
    assertExpected({ chatAvailable: true, leadingMessage: 'blah' });
  });

  it('renders expected leading message if only request callback is available', () => {
    assertExpected({ callbackAvailable: true, leadingMessage: 'blah' });
  });

  it('renders expected leading message if only call us is available', () => {
    assertExpected({ talkAvailable: true, leadingMessage: 'blah' });
  });

  it('renders request callback leading message if request callback and call us are available', () => {
    assertExpected({ callbackAvailable: true, talkAvailable: true, leadingMessage: 'blah' });
  });

  it('renders generic leading message if more than 1 channel is available', () => {
    assertExpected({ chatAvailable: true, talkAvailable: true });
  });

  it('does not render chat if chat is available', () => {
    assertExpected({ submitTicketAvailable: true, oldChat: true });
  });

  it('renders leading message if more than 1 channel is available and useLeadingMessageAsFallback is true', () => {
    assertExpected({
      chatAvailable: true,
      talkAvailable: true,
      leadingMessage: 'hi',
      useLeadingMessageAsFallback: true
    });
  });
});

describe('actions', () => {
  test('clicking on Leave a message', () => {
    const { getByText } = renderComponent({ submitTicketAvailable: true });

    fireEvent.click(getByText('Leave a message'));
    expect(actions.updateActiveEmbed)
      .toHaveBeenCalledWith('ticketSubmissionForm');
    expect(actions.updateBackButtonVisibility)
      .toHaveBeenCalledWith(true);
  });

  test('clicking on Live chat', () => {
    const { getByText } = renderComponent({ chatAvailable: true });

    fireEvent.click(getByText('Live chat'));
    expect(actions.updateActiveEmbed)
      .toHaveBeenCalledWith('chat');
    expect(actions.updateBackButtonVisibility)
      .toHaveBeenCalledWith(true);
  });

  test('clicking on Live chat (old chat)', () => {
    const { getByText } = renderComponent({ oldChat: true, chatAvailable: true });

    fireEvent.click(getByText('Live chat'));
    expect(actions.updateActiveEmbed)
      .toHaveBeenCalledWith('zopimChat');
    expect(actions.updateBackButtonVisibility)
      .toHaveBeenCalledWith(true);
  });

  test('clicking on Request a callback', () => {
    const { getByText } = renderComponent({ callbackAvailable: true, talkAvailable: true });

    fireEvent.click(getByText('Request a callback'));
    expect(actions.updateActiveEmbed)
      .toHaveBeenCalledWith('talk');
    expect(actions.updateBackButtonVisibility)
      .toHaveBeenCalledWith(true);
  });
});
