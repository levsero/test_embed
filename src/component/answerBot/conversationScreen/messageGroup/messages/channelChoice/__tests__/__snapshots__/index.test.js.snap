// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channels does not render chat if chat is available 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      Would you like to leave a message so the team can follow up on your question?
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel singleChannel"
      >
        <span
          class="container icon Icon--channelChoice-contactForm"
          type="Icon--channelChoice-contactForm"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Leave a message
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`channels renders expected leading message if only call us is available 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      blah
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel singleChannel"
      >
        <span
          class="container icon Icon--channelChoice-talk"
          type="Icon--channelChoice-talk"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Call us
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`channels renders expected leading message if only chat is available 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      blah
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel singleChannel"
      >
        <span
          class="container icon Icon--channelChoice-chat"
          type="Icon--channelChoice-chat"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Live chat
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`channels renders expected leading message if only request callback is available 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      blah
    </div>
  </div>
</div>
`;

exports[`channels renders expected leading message if only submit ticket is available 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      blah
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel singleChannel"
      >
        <span
          class="container icon Icon--channelChoice-contactForm"
          type="Icon--channelChoice-contactForm"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Leave a message
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`channels renders generic leading message if more than 1 channel is available 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      Choose a way to get in touch:
    </div>
    <div
      class=" optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-chat"
          type="Icon--channelChoice-chat"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Live chat
        </div>
      </div>
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-talk"
          type="Icon--channelChoice-talk"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Call us
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`channels renders leading message if more than 1 channel is available and useLeadingMessageAsFallback is true 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      hi
    </div>
    <div
      class=" optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-chat"
          type="Icon--channelChoice-chat"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Live chat
        </div>
      </div>
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-talk"
          type="Icon--channelChoice-talk"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Call us
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`channels renders request callback leading message if request callback and call us are available 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      blah
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel singleChannel"
      >
        <span
          class="container icon Icon--channelChoice-talk"
          type="Icon--channelChoice-talk"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Request a callback
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`channels renders the passed in leading message regardless of other properties 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      blah blah
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel singleChannel"
      >
        <span
          class="container icon Icon--channelChoice-contactForm"
          type="Icon--channelChoice-contactForm"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Leave a message
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the expected classes 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      Choose a way to get in touch:
    </div>
    <div
      class=" optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-chat"
          type="Icon--channelChoice-chat"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Live chat
        </div>
      </div>
    </div>
    <div
      class=" optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-talk"
          type="Icon--channelChoice-talk"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Request a callback
        </div>
      </div>
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-contactForm"
          type="Icon--channelChoice-contactForm"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Leave a message
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders the leading message 1`] = `
<div>
  <div
    class="messageBubble container"
  >
    <div
      class="header firstItem title"
    >
      hello world
    </div>
    <div
      class=" optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-chat"
          type="Icon--channelChoice-chat"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Live chat
        </div>
      </div>
    </div>
    <div
      class=" optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-talk"
          type="Icon--channelChoice-talk"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Request a callback
        </div>
      </div>
    </div>
    <div
      class=" lastItem optionItem"
    >
      <div
        class="channel "
      >
        <span
          class="container icon Icon--channelChoice-contactForm"
          type="Icon--channelChoice-contactForm"
        >
          test-file-stub
        </span>
        <div
          class="optionText"
        >
          Leave a message
        </div>
      </div>
    </div>
  </div>
</div>
`;
